### ISSUE TITLE
Need feedback on server connection state

### ISSUE THREAD

**Issue by OlivierS1** [nextcloud/android#616]
### Actual behaviour
-Tell us what happens

I use "Brute force attack" prevention on the server running Nextcloud 11 and sometime it is triggered after several unsuccessful logins.
When Android app is running, it is not possible to tell whether the connection to the server is active.
I usually find those event by loging with the web interface, but it is not convenient.

### Expected behaviour
-Tell us what should happen
 It would be nice to have an indication connection to the server has been lost.

### Steps to reproduce
1. No step in particular except that the firewall will block the network IP. 
2. 
3. 

### Environment data
Android version:
6.0

Device model: 
LG G3

Stock or customized system:
Stock

Nextcloud app version:
1.4.1 and Beta 20170202

Nextcloud server version:
NC 11.0.1

#### Web server error log
```
Insert your webserver log here
```

#### Nextcloud log (data/nextcloud.log)
```
Insert your Nextcloud log here
```

**Comment by tobiasKaminsky** [comment]
@jancborchardt I think I remember that someone wanted to show a little hint on top of the file list if there is no connection. 
Do you think this should also be used for this case?

**Comment by OlivierS1** [comment]
I think that would be just fine as long as it is not based only on Wifi connection alone.
In the current state of the app, when the active page is being refreshed (swiping down), the top portion of the screen shows some activity bar indicating something is happening, but this is not enough to know if the communication with the server is taking place.
A hint, such as a specific icon can be used and possibly needs to be updated only when the app is trying to access the server. At least as a minimum.
Realtime, all the time status update may not be optimal either.

**Comment by tobiasKaminsky** [comment]
My idea was something like that:
When the app figures out that there is no internet connection or it is banned due to brute force protection it should show a warning similar to this picture, but in red.

The warning will disappear if a later connection works (either if the user pull to refresh or changes a folder).

![screenshot_1486201171](https://cloud.githubusercontent.com/assets/5836855/22617476/500704a6-eac6-11e6-982f-485dfeb22000.png)

**Comment by tobiasKaminsky** [comment]
@jancborchardt @eppfel

**Comment by OlivierS1** [comment]
@tobiasKaminsky I do not understand what you are trying to show on the screen capture. Where is the warning you are referring to. Also, are you thinking of changing the color of the app entirely during a no connect event? If so would it be confused with Theming color them?

**Comment by mario** [comment]
@OlivierS1 he would change the message "Showing first 100 results." to something like "No connection" :)

**Comment by tobiasKaminsky** [comment]
@mario exactly, and red 👍

**Comment by eppfel** [comment]
I would be careful with red, it should be clearly visible, but red could conflict, when we introduce theming later on. if the message slides out to grab attention this would suffice IMO.

**Comment by tobiasKaminsky** [comment]
But then we still need a color? Or should it be black font on white background? Then it might be confused with a regular list entry.

**Comment by AndyScherzinger** [comment]
Why not go with grey but add a exclamationmark or info sign in a circle in front of the warning? This way this info would get displayed but is not so "in your face". In the end this is not an error from an app perspective it just means no internet connection, but nothing got broken, corrupted or whatever

**Comment by eppfel** [comment]
> Why not go with grey but add a exclamationmark or info sign in a circle in front of the warning? This way this info would get displayed but is not so "in your face". In the end this is not an error from an app perspective it just means no internet connection, but nothing got broken, corrupted or whatever

This.

 @AndyScherzinger becoming a designer step by step :joy:

It just needs to look different enough to be recognized.

**Comment by mario** [comment]
To me that sounds more like an info than a warning then.

**Comment by AndyScherzinger** [comment]
Yeah, I also think it is rather an info. :+1:

**Comment by tobiasKaminsky** [comment]
To sum up
- if a connection fails show the corresponding warning message
  - too many unsucessfull attempts
  - no internet connection
  - ...
- the warning will stay there as long as a upcoming request to the server is valid
- the warning is grey with an exclamation mark or info sign in a circle in front of the warning and shown directly beneath the header as seen https://github.com/nextcloud/android/issues/616#issuecomment-277432031

correct @eppfel @mario @AndyScherzinger if so please thump up and then label "approved"

**Comment by MorrisJobke** [comment]
I ran into this issue as well. I was connected to a captive portal wifi hotspot and had the app opened. Then I wanted to refresh the files list because I just uploaded something to the server. It just returned from the loading state (those little moving bar below the header) to the normal state without showing any update or error message, but I recognised that all folders were shown as "0 B" in size, which was kind of odd. Then I noticed that the captive portal was blocking it and after clicking through it everything went smooth.
